### PR TITLE
DOC: documentation follow-ups

### PR DESCRIPTION
**Pull request by martinfleis**
A follow-up on #110 

- inheriting proper docstrings
- adding missing methods to API
- moving notebooks from `notebooks` to user guide
- cleaning ReadMe
- pinning numpydoc to avoid a (already fixed) bug in 1.2.0

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -16,6 +15,32 @@ or with financial support from you or your company.
...
+Documentation
+-------------
+
+See the documentation on https://dask-geopandas.readthedocs.io/en/latest/
+
+Installation
+------------
+
+This package depends on GeoPandas and Dask. In addition, it is recommended to
+install PyGEOS, to have faster spatial operations and enable multithreading. See
```
Comment (brendan-ward) [inline README.rst:27]: suggestion:
`In addition, it is recommended to install PyGEOS, to have faster` => `You can also install PyGEOS to have faster...`

Comment (jorisvandenbossche) [reply README.rst:27]: I would maybe phrase it a bit stronger than "you can also install", as it is really strongly recommended no? (I don't think there is much point in using dask-geopandas without it ..)

File: doc/source/guide/basic-intro.ipynb [hunk 2]
```diff
@@ -17,17 +17,14 @@
     "import geopandas\n",
     "\n",
     "import dask.dataframe as dd\n",
-    "import dask_geopandas\n",
-    "\n",
-    "import pyproj\n",
-    "pyproj.datadir.set_data_dir(\"/home/joris/miniconda3/envs/geo-dev/share/proj/\")\n",
```
Comment (jorisvandenbossche) [inline doc/source/guide/basic-intro.ipynb:23]: Good catch! :)

File: README.rst (hunk 1 above)
Comment (martinfleis) [reply README.rst:27]: Well, this sentence is incorrect anyway. We depend on `pygeos` in `calculate_spatial_partitions`.

**Pull request merged**